test/e2e/autoscaling_test.go

```diff
@@ -96,9 +98,13 @@ func TestAutoscaling(t *testing.T) {
 
 	// Validate nodepools are scaled out and balanced.
 	// Each nodepool should have 1 more node.
+	// we only check that machineDeployments replicas are as expected to avoid timeout waiting for Ready condition on the nodes.
+	expectedNodesNum := min + 1
 	for _, nodepool := range nodepools.Items {
-		expectedNodesNum := min + 1
-		_ = e2eutil.WaitForNReadyNodesByNodePool(t, ctx, guestClient, expectedNodesNum, hostedCluster.Spec.Platform.Type, nodepool.Name)
+		md, err := e2eutil.MachineDeploymentByNodepool(ctx, client, hostedCluster, nodepool.Name)
+		g.Expect(err).ToNot(HaveOccurred(), "failed to get machindeDeployment for nodePool %s", nodepool.Name)
+
+		g.Expect(md.Spec.Replicas).To(Equal(expectedNodesNum))
```

Comment:
This needs to be in a poll function as it takes a while for autoscaler to scale out. The way it's now it might/might not pass.
You'll want to do the same for scale in validation in line 119

Reply:
true, that makes sense.
`WaitForNReadyNodes` func in line 119 already includes a poll.